File: src/main/java/com/github/codingvampyre/drachenfest/domain/Ticket.java
package com.github.codingvampyre.drachenfest.domain;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@AllArgsConstructor
@Builder
public class Ticket extends BaseEntity {

    @NotNull
    private String barcode;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "event_id")
    private Event event;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ticket_type_id")
    private TicketType ticketType;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "account_id")
    private Account account;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "purchase_id")
    private Purchase purchase;
}

File: src/main/java/com/github/codingvampyre/drachenfest/domain/Ticket.kt
package com.github.codingvampyre.drachenfest.domain

import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
class Ticket : BaseEntity() {
	@Column(nullable = false)
	private val barcode: @NotNull String? = null

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "event_id")
	private val event: Event? = null

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "ticket_type_id")
	private val ticketType: TicketType? = null

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "account_id")
	private val account: Account? = null

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "purchase_id")
	private val purchase: Purchase? = null
}